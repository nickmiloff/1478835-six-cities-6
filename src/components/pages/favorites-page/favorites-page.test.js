import * as React from 'react';
import {render} from '@testing-library/react';
import Test from '../../test/test';

import FavoritesPage from './favorites-page';

describe(`FavoritesPage component test`, () => {
  it(`'FavoritesPage component' should render correctly with no cards`, () => {
    const {container} = render(
        <Test>
          <FavoritesPage />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });

  it(`'FavoritesPage component' should render correctly with cards`, () => {
    const {container} = render(
        <Test initial={false}>
          <FavoritesPage />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
