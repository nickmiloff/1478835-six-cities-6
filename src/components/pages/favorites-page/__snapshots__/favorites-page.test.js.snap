// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage component test 'FavoritesPage component' should render correctly with cards 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    mail@gmail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <button
                  class="header__nav-link header__nav-link--profile"
                  style="background-color: transparent;"
                >
                  <span
                    class="header__login"
                  >
                    Logout
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites"
        >
          <h1
            class="favorites__title"
          >
            Saved listing
          </h1>
          <ul
            class="favorites__list"
          >
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Paris
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places"
              >
                <article
                  class="favorites__card place-card"
                >
                  
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/3"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          865
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button button place-card__bookmark-button--active"
                        data-testid="to-bookmarks-button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 92%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/3"
                      >
                        Tile House
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      house
                    </p>
                  </div>
                </article>
                <article
                  class="favorites__card place-card"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/10"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          156
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button button place-card__bookmark-button--active"
                        data-testid="to-bookmarks-button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 56%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/10"
                      >
                        Amazing and Extremely Central Flat
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      house
                    </p>
                  </div>
                </article>
              </div>
            </li>
            
            
            
            
            
          </ul>
        </section>
      </div>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;

exports[`FavoritesPage component test 'FavoritesPage component' should render correctly with no cards 1`] = `
<div>
  <div
    class="page page--favorites-empty"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites page__main--favorites-empty"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites favorites--empty"
        >
          <h1
            class="visually-hidden"
          >
            Favorites (empty)
          </h1>
          <div
            class="favorites__status-wrapper"
          >
            <b
              class="favorites__status"
            >
              Nothing yet saved.
            </b>
            <p
              class="favorites__status-description"
            >
              Save properties to narrow down search or plan your future trips.
            </p>
          </div>
        </section>
      </div>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;
